src/slanderer/RobotPlayer.java

```diff
@@ -0,0 +1,93 @@
+package examplefuncsplayer;
+import battlecode.common.*;
+
+public strictfp class RobotPlayer {
+    static RobotController rc;
+
+    /**
+     * Method run when robot spwans in world
+     **/
+    @SuppressWarnings( "unused" )
+    public static void run( RobotController rc ) throws GameActionException {
+
+        // Instance of robot for control actions
+        RobotPlayer.rc = rc;
+
+        while ( true ) {
+            // Try/catch blocks stop unhandled exceptions, which cause your robot to freeze
+            try {
+                // Who am I?
+                switch ( rc.getType() ) {
+                    case SLANDERER:     runSlanderer();     break;
```

Comment:
If this code only gets loaded for slanderers (Is this how it works? I am unclear on what code gets loaded based on the getting started guide.) then I shouldn't even need a case stmt. I'll have to see how the game loads these files into the robots.

Reply:
Right ok so I think this is becoming clearer.
When you run the game you can supply options: `./gradlew run -PteamA=basicplayer -PteamB=slanderer`
so this makes my slanderer RobotPlayer run for all of teamB robots...

Reply:
I think this is right, the idea being that we pass the enlightenment center as our bot? And then that builds the others?

Reply:
So I think basically the idea is we have only one subfolder called `final_robot` or `ultimate_bot` or whatever and in that we have `RobotPlayer.java` which has the basic template code when gets loaded on to each robot no matter what type it is.

Then we hit that run method with the case stmt to check what type of robot we are from the current running instance's perspective. This I guess then calls out to our robot classes like mukraker etc.

If that sounds right I'll redo this branch to fit that style.